internal/terraform/version_required.go

```diff
@@ -27,6 +27,25 @@ func CheckCoreVersionRequirements(config *configs.Config) tfdiags.Diagnostics {
 	module := config.Module
 
 	for _, constraint := range module.CoreVersionConstraints {
+		// Before checking if the constraints are met, check that we are not using any prerelease fields as these
+		// are not currently supported.
+		for _, required := range constraint.Required {
+			if required.Prerelease() {
+				diags = diags.Append(&hcl.Diagnostic{
+					Severity: hcl.DiagError,
+					Summary:  "Invalid required_version constraint",
+					Detail:   fmt.Sprintf("Prerelease version constraints are not supported: %s.", required.String()),
```

Comment:
I'm wondering if this should also suggest the fix of removing the prerelease information and that the version constraint may still match the prerelease?